scripts/dev/newReleaseBranch.sh

```diff
@@ -0,0 +1,681 @@
+
+Examples:
+  1. After the release of 23.1.0 from the master branch:
+     $ $0 23.1.0
+     Creates the HotFix branch for 23.1.0, creates the 23.1.N ServicePack branch and modifies the master branch for
+     use by the next minor version 23.2.0.
+
+  2. After the release of 23.1.1 from the ServicePack branch release/23.1.N:
+     $ $0 23.1.1
+     Creates the HotFix branch for 23.1.1 and modifies the 23.1.N ServicePack branch for use by the next version 23.1.2
+
+  3. After the release of 23.1.2 from the ServicePack branch release/23.1.N
+     $ $0 23.1.2
+     Creates the HotFix branch for 23.1.2 and modifies the 23.1.N ServicePack branch for use by the next version 23.1.3
+
+  4. Switching to the next major version:
+     $ $0 23.2.0 -v 25.1.0
+     Creates the HotFix branch for 23.2.0, creates the 23.2.N ServicePack branch and modifies the master branch for
+     use by the next Hot Fix major version 25.1.0.
+???
+}
+
+readCommandLineArgs() {
+  hotFixVersion="${1}"
+  if [ -z "${hotFixVersion}" ]
+  then
+    usage
+    exit 1
+  fi
+  shift 1
+
+  masterVersion=""
+  doPush="true"
+  doTest=""
+  doCleanup=""
+  loggingOut="/dev/null"
+  prefix=""
+  while getopts "v:stclh" arg; do
+      case $arg in
+          v)
+              masterVersion=${OPTARG}
+              ;;
+          l)
+              loggingOut=`tty`
+              prefix="== "
+              ;;
+          t)
+              doTest="true"
+              ;;
+          c)
+              doCleanup="true"
+              ;;
+          s)
+              doPush=""
+              ;;
+          h | *)
+              usage
+              exit 0
+              ;;
+      esac
+  done
+  shift $((OPTIND-1))
+
+  if [ "$#" != "0" ]; then
+    usage
+    exit 1
+  fi
+}
+
+getVersionMajor() {
+  echo "${1}" | sed -n "s/^\([0-9]*\)\.[0-9]*\.[0-9]*$/\1/p"
+}
+
+getVersionMinor() {
+  echo "${1}" | sed -n "s/^[0-9]*\.\([0-9]*\)\.[0-9]*$/\1/p"
+}
+
+getVersionRevision() {
+  echo "${1}" | sed -n "s/^[0-9]*\.[0-9]*\.\([0-9]*\)$/\1/p"
+}
+
+increment() {
+  local number="${1}"
+  echo "${number} + 1" | bc
+}
+
+function readTopLevelTag() {
+  local tagName="${1}"
+  local pomFile="${2}"
+  # Might be possible to generalise this function to accept an XPath so it could be used in place of sed commands
+
+  # Read the file with an IFS (Input Field Separator) of the start of XML tag character <
+  local IFS=\>
+  local depth=-99
+  while read -d \< ENTITY CONTENT
+  do
+    if [[ $ENTITY == project\ * ]] # outer <project> tag
+    then
+      depth=0
+    elif [[ $ENTITY == /* ]] # end tag
+    then
+      ((depth=depth-1))
+    else                     # start tag
+      ((depth=depth+1))
+    fi
+
+    if [[ $ENTITY = "${tagName}" ]] && [[ $depth == 1 ]] ; then
+        echo $CONTENT
+        exit
+    fi
+  done < $pomFile
+  exit 1
+}
+
+getPomVersion() {
+  # Same as slower/simpler: "mvn help:evaluate -Dexpression=project.version"
+  readTopLevelTag version pom.xml
+}
+
+getPomMajor() {
```

Comment:
I was thinking about passing the version once as a parameter to these 3 methods, but it's probably negligible in performance while allowing cleaner usage overall